hot/style-loader.js

```diff
@@ -20,7 +20,21 @@ module.exports.pitch = function (request) {
       const updatePanelElems = require('panel/hot/update-panel-elems');
       updatePanelElems('${elemName}', elem => {
         if (elem.getConfig('useShadowDom')) {
-          elem.el.querySelector('style').textContent = newStyle.toString();
+          const newStyleText = newStyle.toString();
+          if (elem.el.adoptedStyleSheets) {
+            if (!elem.configStyleSheet) {
+              elem.configStyleSheet = new CSSStyleSheet();
+              elem.el.adoptedStyleSheets = [elem.configStyleSheet, ...this.el.adoptedStyleSheets.slice(1)];
+            }
+            elem.configStyleSheet.replaceSync(newStyleText);
+            elem.setCachedStyleSheet(elem.configStyleSheet);
+          } else {
+            if (!elem.configStyleTag) {
+              elem.configStyleTag = document.createElement('style');
+              elem.el.prepend(elem.configStyleTag);
+            }
+            elem.configStyleTag.textContent = newStyleText;
+          }
```

Comment:
This is so close to the component code, but not exactly the same. It feels like there must be an opportunity here to make a single publicly-accessible component method that you can use for setting style both in the component code and the hot loader. WDYT?

Reply:
Makes sense, I've made an idempotent `Component.applyStaticStyle()` method that should do the right thing.